app/models/user.rb

```diff
@@ -22,6 +22,10 @@ class User < ActiveRecord::Base
   has_many :courses, through: :course_users
   has_many :course_group_users, dependent: :destroy, class_name: Course::GroupUser.name
   has_many :course_groups, through: :course_group_users, class_name: Course::Group.name
+  has_many :topic_subscriptions, class_name: Course::Discussion::Topic::Subscription.name,
```

Comment:
Without these relations, there will be `Could not find the inverse association` exception. And we need `belongs_to` associations in `Subscription` related models. But it seems we can not define these associations alone.